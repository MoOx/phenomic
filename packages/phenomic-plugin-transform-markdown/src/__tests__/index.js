import transformMarkdown from "../index.js"

it("should transform markdown as html", () => {
  const plugin = transformMarkdown()
  expect(typeof plugin.transform === "function").toBe(true)
  expect(
    plugin.transform &&
      plugin.transform({
        file: {
          name: "file.json",
          fullpath: "/test/file.json",
          exists: true,
          type: "wat",
        },
        contents: new Buffer(
          `---
test: a
test2: b
---
` + "## Test\n[link](href)\n```js\nconsole.log(window)\n```",
        ),
      }),
  ).toMatchSnapshot()
})
