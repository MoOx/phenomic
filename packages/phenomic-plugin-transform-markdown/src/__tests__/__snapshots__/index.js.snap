// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should transform markdown as html 1`] = `
Object {
  "data": Object {
    "body": "<h2 id=\\"test\\"><a href=\\"#test\\" class=\\"phenomic-HeadingAnchor\\">#</a>Test</h2>
<p><a href=\\"href\\">link</a></p>
<pre><code class=\\"language-js\\">console.log(window)
</code></pre>
",
    "test": "a",
    "test2": "b",
  },
  "partial": Object {
    "test": "a",
    "test2": "b",
  },
}
`;
